frontend/packages/console-shared/src/hooks/__tests__/usePinnedResources.spec.ts

```diff
@@ -86,7 +96,29 @@ describe('usePinnedResources', () => {
     expect(setPinnedResourcesMock).toHaveBeenCalledTimes(0);
   });
 
+  it('returns customized pins if the pins are not customized by the user and the extension has default pins', async () => {
```

Comment:
Nice!

Can you add some more tests here?

As I wrote above the other tests should check if the old code still works fine. Here I would like to see also cases like:

1.

```
  it('returns default pins if the perspective is configured without pinned resources', async () => {
    window.SERVER_FLAGS.perspectives = '[{ "id" : "dev", "visibility": {"state" : "Enabled" }}]';
```

2.

```
  it('returns no pins if the perspective is configured with an empty array', async () => {
    window.SERVER_FLAGS.perspectives = '[{ "id" : "dev", "visibility": {"state" : "Enabled" }, "pinnedResources": [] }]';
```

3. then your tests

4. and then one more with a typo. If this return the typo here, that's fine. But then we have a documentation that the hook returns it and the frontend should filter it:

```
  it('returns no pins if the perspective is configured with an empty array', async () => {
    window.SERVER_FLAGS.perspectives =
      '[{ "id" : "dev", "visibility": {"state" : "Enabled" }, "pinnedResources" : [{"version" : "v1", "resource" : "deployments", "group": "apps"}, {"version" : "v1", "resource" : "invalid-resource", "group": ""}]}]';
}]';
```